server.js

```diff
@@ -1,23 +1,99 @@
 const express = require("express");
 const cors = require("cors");
+const bodyParser = require("body-parser");
 
 const app = express();
 
-app.use(cors());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+const port = 3007;
+
+app.use(
+  cors({
+    origin: "http://localhost:3007",
+  })
+);
 
 const welcomeMessage = {
   id: 0,
-  from: "Bart",
+  from: "Pakize",
   text: "Welcome to CYF chat system!",
+  timeSpent: "24 / 12 / 2022"
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+
+//Read only text whose text contains a given substring
+
+app.get("/messages/search", (req, res) => {
+  const { term } = req.query;
+  console.log(term);
+
+  const filterMessages = messages.filter((message) =>
+    message.text.toLowerCase().includes(term.toLowerCase())
+  );
+  console.log(filterMessages);
+  res.send(filterMessages);
 });
 
-app.listen(process.env.PORT);
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+// All messages
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+//  Read only the last 10 messages:
+
+app.get("/messages/latest",(req,res)=>{
+  res.send(messages.slice(-10))
+})
+
+// A new message
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+  // console.log(req.body)
+  const ourMessageObject = {
+    id: messages.length + 1,
```

Comment:
this method of selecting an id can lead to duplicate id values.

Imaging you had a set of 20 messages, where the last message is id:20. You delete message 19. When you next send a message, it will also be given id:20